pf/proto/internal/gather/attr.go

```diff
@@ -0,0 +1,84 @@
+// Copyright 2016-2024, Pulumi Corporation.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package gather
+
+import (
+	"github.com/hashicorp/terraform-plugin-framework/attr"
+	"github.com/hashicorp/terraform-plugin-go/tfprotov6"
+
+	"github.com/pulumi/pulumi-terraform-bridge/pf/internal/pfutils"
+)
+
+type _attr struct{ s *tfprotov6.SchemaAttribute }
+
+var _ = pfutils.Attr(_attr{})
```

Comment:
I'm wondering if there is any way to avoid layering these adapters on top of pfutils adapters.

pfutils.Attr is a workaround as it doc comment states

There is some additional highly suspect reflective code in:

https://github.com/pulumi/pulumi-terraform-bridge/blob/master/pf/internal/pfutils/reflector.go#L66

A "clean" adapter set would go from the Protocol V6 Schema encoding straight into shim.Schema. 

Is that not practical?

Reply:
> A "clean" adapter set would go from the Protocol V6 Schema encoding straight into shim.Schema.
>
>Is that not practical?

I think it's practical, but it would force this PR to reach much farther into `/pf` then I'd like. I'd rather merge with the current `pfutils` setup and then refactor afterwards.

Reply:
`pfutils` has been removed here.